`lit/no-this-assign-in-render` seems to false positive when you assign a local variable with the value of this.property

fine:
```
const _twitter: TemplateResult | symbol = this.twitter ? html`<li><a href=${`https://twitter.com/${this.twitter}`} target="_blank" rel="noreferrer noopener">Twitter</a></li>` : nothing;
```

fails:
```
let _twitter: TemplateResult | symbol = nothing;
if (this.twitter) {
  _twitter = html`<li><a href=${`https://twitter.com/${this.twitter}`} target="_blank" rel="noreferrer noopener">Twitter</a></li>`; // eslint warns
}
```

https://gist.github.com/halkeye/d89a183a92f082ccd76ea313a98c81c4

```
 npx eslint src/jio-socialmedia-buttons.ts

/home/halkeye/go/src/github.com/jenkins-infra/jenkins-io-components/src/jio-socialmedia-buttons.ts
  27:7  error  Members of `this` should not be assigned to in the render method. It is likely you should do this elsewhere instead (e.g. in `updated`)  lit/no-this-assign-in-render

✖ 1 problem (1 error, 0 warnings)
```

--- a/src/rules/no-this-assign-in-render.ts
+++ b/src/rules/no-this-assign-in-render.ts
@@ -114,7 +114,7 @@ const rule: Rule.RuleModule = {
       MethodDefinition: (node: ESTree.Node): void =>
         methodEnter(node as ESTree.MethodDefinition),
       'MethodDefinition:exit': methodExit,
-      'AssignmentExpression:has([left] ThisExpression)': (
+      'AssignmentExpression:has(.left ThisExpression)': (
         node: ESTree.Node
       ): void => assignmentFound(node as ESTree.AssignmentExpression)
     };

--- a/src/test/rules/no-this-assign-in-render_test.ts
+++ b/src/test/rules/no-this-assign-in-render_test.ts
@@ -49,6 +49,18 @@ ruleTester.run('no-this-assign-in-render', rule, {
         static render() {
           this.foo = 5;
         }
+      }`,
+    `class Foo extends LitElement {
+        render() {
+          let x;
+          x = this.prop;
+        }
+      }`,
+    `class Foo extends LitElement {
+        render() {
+          let x;
+          x = 5;
+        }
       }`
   ],
 

